libs/core/langchain_core/tools.py

```diff
@@ -1218,3 +1218,32 @@ class BaseToolkit(BaseModel, ABC):
     @abstractmethod
     def get_tools(self) -> List[BaseTool]:
         """Get the tools in the toolkit."""
+
+
+def _get_description_from_runnable(runnable: Runnable) -> str:
+    """Generate a placeholder description of a runnable."""
+    input_schema = runnable.input_schema.schema()
+    return f"Takes {input_schema}."
+
+
+def _get_args_schema_for_runnable(runnable: Runnable) -> Type[BaseModel]:
+    """Get object schema from runnable input schema."""
+    schema_name = f"{runnable.get_name()}Schema"
+    input_name = "input"
```

Comment:
Are we able to pick the appropriate input-name based on the runnable sequence?

Reply:
newest implementation does this